### PR TITLE
Add logging for spicedb connection on startup

### PR DESCRIPTION
**Pull request by lennysgarage**
## Describe your changes

- Makes requests to SpiceDB during startup, exits if no connection within 30s
- - Currently, service restarts indefinitely and tries to reconnect to SpiceDB for another 30s.
- Request validation for check requests.
- Takes advantage of `WaitForReady(true)` to [block until a connection to spicedb is achieved](https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#:~:text=default%2C%20RPCs%20will,to%20a%20server). 

## Ticket reference (if applicable)
Fixes # https://issues.redhat.com/browse/RHCLOUD-33335

## Checklist

* [ ] Are the agreed upon acceptance criteria fulfilled?

* [ ] Was the 4-eye-principle applied? (async PR review, pairing, ensembling)

* [ ] Do your changes have passing automated tests and sufficient observability?

* [ ] Are the work steps you introduced repeatable by others, either through automation or documentation?
  * [ ] If automation is possible but not done due to other constraints, a ticket to the tech debt sprint is added
  * [ ] An SOP (Standard Operating Procedure) was created

* [ ] The Changes were automatically built, tested, and  - if needed, behind a feature flag - deployed to our production environment. (**Please check this when the new deployment is done and you could verify it.**)

* [ ] Are the agreed upon coding/architectural practices applied?

* [ ] Are security needs fullfilled? (e.g. no internal URL)

* [ ] Is the corresponding Ticket in the right state? (should be on "review" now, put to done when this change made it to production)

* [ ] For changes to the public API / code dependencies: Was the whole team (or a sufficient amount of ppl) able to review?

### CODE REVIEW

File: internal/data/spicedb.go [hunk 1]
```diff
@@ -70,6 +71,22 @@ func NewSpiceDbRepository(c *conf.Data, logger log.Logger) (*SpiceDbRepository,
 		return nil, nil, err
 	}
 
+	// wait upto 30 seconds to confirm spicedb is connected
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*30)
+	defer cancel()
+
+	_, err = client.ReadSchema(ctx, &v1.ReadSchemaRequest{}, grpc.WaitForReady(true))
+	if err != nil {
+		return nil, nil, fmt.Errorf("error trying to reach SpiceDB: %w", err)
```
Comment (Rajagopalan-Ranganathan) [inline internal/data/spicedb.go:80]: @wscalf and myself dicussed this a bit, it wouldnt be ideal to return a error here.

Comment (lennysgarage) [reply internal/data/spicedb.go:80]: After discussing with @wscalf, we agreed that due to https://github.com/project-kessel/relations-api/pull/111, that adding logging/erroring here at all is redundant and not very useful. This only tells us the connection state at one point in time, while the work done with the readyz endpoint allows to check the state at any point.

Comment (Rajagopalan-Ranganathan) [reply internal/data/spicedb.go:80]: I assume we can close this then @lennysgarage

Comment (lennysgarage) [reply internal/data/spicedb.go:80]: Yes, I'll go ahead with it

**Comment by lennysgarage** [comment]
Closed due to https://github.com/project-kessel/relations-api/pull/111

**Pull request closed without merging**